code/src/scripts/job_distributor_deployment_state.py

```diff
@@ -20,10 +28,47 @@ def _set_command_specific_options(self, parser):
         parser.add_argument('--interval', dest='interval', metavar='INTERVAL',
                             default=self.collect_interval, type=int, help=hmsg)
 
+    def _with_parent(self, active, with_parent):
+        for r in active.resources:
+            p_id = r.data['parent']
+            try:
+                self.api.get(p_id, select='')
+                with_parent.append(r)
+            except NuvlaError as ex:
+                logging.warning(f'Parent {p_id} is missing for {r.id}')
+
+    def _old_deployments(self):
+        return self.collect_interval > COLLECT_INTERVAL_SHORT
+
+    def _publish_metric(self, name, value):
+        interval = 'old' if self._old_deployments() else 'new'
+        mname = f'job_distributor.{self.ACTION_NAME}.{interval}.{name}'
+        super()._publish_metric(mname, value)
+
     def active_deployments(self):
-        return map(lambda x: x.id,
-                   self.api.search('deployment',
-                                   filter="state='STARTED'", select='id').resources)
+        # Collect old or new deployments.
+        past = (datetime.now() - timedelta(seconds=COLLECT_PAST_SEC))\
```

Comment:
Here you are using the time of Job-engine. I would use the filter using the time of the ES because it's the reference by using the query this way and it's clearer:
filters = f"state='STARTED' and updated<'now-{COLLECT_PAST_SEC}s'"

Reply:
Agreed.

Reply:
Fix pushed in 6ca8452